secret-mode.el

```diff
@@ -27,15 +27,31 @@
 
 ;;; Code:
 
-(defconst secret-mode-table
-  (let ((disptbl (make-display-table)))
-    (dotimes (i 1024)
-      (aset disptbl i
-            (pcase (get-char-code-property i 'general-category)
-              ((or 'Cc 'Cf 'Zs 'Zl 'Zp) nil)
-              ('Lu (vector (make-glyph-code ?▆)))
-              (_   (vector (make-glyph-code ?▃))))))
-    disptbl)
+(eval-and-compile
```

Comment:
I'm quite new on Emacs Lisp, what does `eval-and-compile` do?

Reply:
It makes the definitions enclosed by `eval-and-compile` be accessible during the compilation. This allows for pre-computing of the whole display table at compile time, so there is no performance hit when the package is being loaded. Also, see https://www.gnu.org/software/emacs/manual/html_node/elisp/Eval-During-Compile.html